src/main/java/org/javarosa/core/model/condition/EvaluationContext.java

```diff
@@ -297,15 +322,31 @@ private void expandReferenceAccumulator(TreeReference sourceRef, DataInstance so
 
         AbstractTreeElement node = sourceInstance.resolveReference(workingRef);
 
+        this.openBulkTrace();
+
         // Use the reference's simple predicates to filter the potential
         // nodeset.  Predicates used in filtering are removed from the
         // predicate input argument.
-        Vector<TreeReference> childSet = node.tryBatchChildFetch(name, mult, predicates, this);
+        Collection<TreeReference> childSet = node.tryBatchChildFetch(name, mult, predicates, this);
+
+        this.reportBulkTraceResults(originalPredicates, predicates, childSet);
```

Comment:
Predicates is mutable and gets incrementally shaved down as predicates are "batch fetched". OriginalPredicates contains the predicates that were requested coming into the method.